tasks/configure_ssh_keys.yml

```diff
@@ -1,103 +1,64 @@
 ---
+- name: Get ssh keys for jenkins slave
+  block:
+  - name: "Create a directory .ssh if it doesn't exist"
+    file:
+      path: "{{ jenkins2_home_directory }}/.ssh"
+      state: directory
+      mode: 0700
 
-- name: Create a directory .ssh if it doesn't exist
-  file:
-    path: "{{ jenkins2_home_directory }}/.ssh"
-    state: directory
-    mode: 0700
-  become: true
+  - name: "Ensure that openssh *private* key doesn't exist"
+    stat:
+      path: "{{ jenkins2_home_directory }}/.ssh/\
+      {{ jenkins2_ssh_keys_private_keyname }}"
+    register: private_key_stat
 
-- name: Ensure that openssh *private* key doesn't exist
-  stat:
-    path: "{{ jenkins2_home_directory }}/.ssh/\
-    {{ jenkins2_ssh_keys_private_keyname }}"
-  become: true
-  register: private_key_stat
+  - name: "Ensure that openssh *public* key doesn't exist"
+    stat:
+      path: "{{ jenkins2_home_directory }}/.ssh/\
+      {{ jenkins2_ssh_keys_private_keyname }}.pub"
+    register: public_key_stat
 
-- name: Ensure that openssh *public* key doesn't exist
-  stat:
-    path: "{{ jenkins2_home_directory }}/.ssh/\
-    {{ jenkins2_ssh_keys_private_keyname }}.pub"
-  become: true
-  register: public_key_stat
+  - name: "Generate a pair of keys"
+    command: >-
```

Comment:
use modules please

Reply:
Module openssl_privatekey don't properly work for jenkins credential. In ansible 2.8 have a module named openssh_keypair, but we have an issue with work molecule and ansible 2.8. So, for now, command module its workaround these problems.